packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -0,0 +1,34 @@
+import type { MetaMaskProvider } from '@consensys/get-starknet/dist/type';
+import detectEthereumProvider from '@metamask/detect-provider';
+
+async function hasSnapSupport(provider: any) {
+  try {
+    await provider.request({
+      method: "wallet_getSnaps",
+    })
+    return true
+  } catch {
+    return false
+  }
+}
+
+async function bootstrapMetamaskBridge() {
+  if (window.hasOwnProperty("starknet_metamask")) {
+    return;
+  }
+
+  const provider = await detectEthereumProvider<MetaMaskProvider>()
+  if (!provider) {
+    return;
+  }
+
+  const snapSupport = await hasSnapSupport(provider)
+  if (!snapSupport) {
+    return;
+  }
+
+  const {MetaMaskSnapWallet} = await import('@consensys/get-starknet')
+  window["starknet_metamask"] = new MetaMaskSnapWallet(provider)
+}
+
+export {bootstrapMetamaskBridge}
```

Comment:
nit:
end of line spacking

Reply:
resolved

Reply:
Done (file has changed but I verified the formatting)